### PR TITLE
add some docker capabilities

### PR DESCRIPTION
**Pull request by fernandoneto**
- Remove images that are not being used
- Remove all stoped container's

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 ## Upcoming
 
+## 1.0.3
```
Comment (cusspvz) [inline CHANGELOG.md]: please remove the tag from here, It should be always as upcoming until it is released

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,8 @@
 FROM alpine:edge
 MAINTAINER José Moreira <jose.moreira@findhit.com>
 ADD tunesys /tunesys
-RUN chmod +x /tunesys/run /tunesys/common /tunesys/tuner/*
+RUN apk add --update docker; \
```
Comment (cusspvz) [inline Dockerfile:4]: Please place into a separated `RUN` statement and don't forget to purge apk cache.

File: Dockerfile [hunk 3]
```diff
@@ -19,5 +20,13 @@ ENV TUNE_TCP=1 \
     net__ipv4__tcp_tw_reuse=1 \
     net__ipv4__tcp_max_orphans=0
 
+# DOCKER_IMAGES
+ENV TUNE_IMAGES=1 \
```
Comment (cusspvz) [inline Dockerfile]: I would prefer `TUNE_DOCKER` instead and leave the containers/images cleanup as inner options.

File: Dockerfile [hunk 4]
```diff
@@ -1,6 +1,10 @@
 FROM alpine:edge
 MAINTAINER José Moreira <jose.moreira@findhit.com>
 ADD tunesys /tunesys
+RUN apk add --update docker; \
+    rm -rf \
+    /etc/ssl \
```
Comment (cusspvz) [inline Dockerfile]: Is there a specific reason that lead you to remove ssl folder?

File: Dockerfile [hunk 5]
```diff
@@ -19,5 +23,9 @@ ENV TUNE_TCP=1 \
     net__ipv4__tcp_tw_reuse=1 \
     net__ipv4__tcp_max_orphans=0
 
+# DOCKER_IMAGES
+ENV TUNE_DOCKER=1 \
+    docker_loop_interval=21600
+
```
Comment (cusspvz) [inline Dockerfile:29]: Add options to allow enabling/disabling of the removing (images/stopped_containers) features

**Comment by fernandoneto** [comment]
@cusspvz can you take a look now?

File: Dockerfile [hunk 6]
```diff
@@ -19,5 +23,11 @@ ENV TUNE_TCP=1 \
     net__ipv4__tcp_tw_reuse=1 \
     net__ipv4__tcp_max_orphans=0
 
+# DOCKER_IMAGES
+ENV TUNE_DOCKER=1 \
+    R_CONTAINERS=false \
```
Comment (cusspvz) [inline Dockerfile]: `docker_remove_containers=1`

File: Dockerfile [hunk 7]
```diff
@@ -19,5 +23,11 @@ ENV TUNE_TCP=1 \
...
+    R_IMAGES=false \
```
Comment (cusspvz) [inline Dockerfile]: `docker_remove_images=1`

File: Dockerfile (hunk 4 above)
Comment (cusspvz) [inline Dockerfile]: remove this line please

File: README.md [hunk 8]
```diff
@@ -19,14 +19,20 @@ It has built-in configurable tuners that handles minor settings, such as:
     - Replaces hand `sysctl ...` configurations
     - Tunes settings every hour
     - This isn't One Shot, it will act every hour (interval also configurable)
+  * Docker
+    - Removes all images that are not being used. Option disable by default `R_IMAGES=false` in Dockerfile
```
Comment (cusspvz) [inline README.md]: `docker_remove_images`

File: README.md [hunk 9]
```diff
@@ -19,14 +19,20 @@ It has built-in configurable tuners that handles minor settings, such as:
...
+    - Removes all stoped containers. Option disable by default `R_CONTAINERS=false` in Dockerfile.
```
Comment (cusspvz) [inline README.md]: `docker_remove_containers`

File: README.md [hunk 10]
```diff
@@ -19,14 +19,20 @@ It has built-in configurable tuners that handles minor settings, such as:
...
 
 ## Deploying
 
 ```bash
 docker run -d \
-    --privileged \ # to tune kernel settings 
+    --privileged \ # to tune kernel settings
     --net host \ # to tune host network
+    --volume /var/run/docker.sock:/var/run/docker.sock \
+    --env IMAGES=true \ # to remove docker images
```
Comment (cusspvz) [inline README.md]: `docker_remove_images`

File: README.md [hunk 11]
```diff
@@ -19,14 +19,20 @@ It has built-in configurable tuners that handles minor settings, such as:
...
+    --env CONTAINERS=true \ # to remove docker container's
```
Comment (cusspvz) [inline README.md]: `docker_remove_containers`

File: tunesys/tuner/DOCKER [hunk 12]
```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+
+while true; do
+
+  if [ $R_CONTAINERS == true ]; then
```
Comment (cusspvz) [inline tunesys/tuner/DOCKER]: `$docker_remove_images == "1"`

File: tunesys/tuner/DOCKER [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+
+    echo "Removing docker stoped container's "
+    docker ps -aq --no-trunc | xargs docker rm > /dev/null 2>&1
+
+  else
+
+    echo "You choose do not remove stoped docker container's"
+
+  fi
```
Comment (cusspvz) [inline tunesys/tuner/DOCKER:10]: remove the entire else statement

File: tunesys/tuner/DOCKER [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+
+  if [ $R_IMAGES == true ]; then
+
+    echo "Removing docker images that are not being used"
+    docker images -q | xargs docker rmi > /dev/null 2>&1
+
+  else
+
+    echo "You choose do not remove docker images"
```
Comment (cusspvz) [inline tunesys/tuner/DOCKER]: remove the entire else statement

**Comment by cusspvz** [comment]
Thanks!!

**Pull request merged**